### PR TITLE
Use custom contract pages for token contracts

### PR DESCRIPTION
**Pull request by nachoiacovino**

### CODE REVIEW

**Comment by jnsdls** [comment]
#### 📝 Changed routes:


| [**/dashboard**](https://thirdweb-adfn38e0z-thirdweb.vercel.app/dashboard) |
|:-:|
|<a href="https://thirdweb-adfn38e0z-thirdweb.vercel.app/dashboard">Screenshot of /dashboard</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/2gLkg7hjCxBDFuW4D7jpO2qMjd55NWCUKpA9rJRfuN5wZWqwiinShiGaVNYKc0wLvyqcLweAK23oqGJFEBLP1Q.png">(view full size)</a> |







Commit 85b3bf8fbe41fdc16333b398a65d8ef0cc62771a (https://thirdweb-adfn38e0z-thirdweb.vercel.app).

File: constants/mappings.ts [hunk 1]
```diff
@@ -24,19 +24,19 @@ export const FeatureIconMap: Record<ContractType, StaticImageData> = {
...
-  split: "split",
-  "token-drop": "token-drop",
+  split: "/split",
+  "token-drop": "/token-drop",
   // TODO (byoc)
   custom: "",
-  "signature-drop": "signature-drop",
+  "signature-drop": "/signature-drop",
   multiwrap: "",
 };
```
Comment (jnsdls) [inline constants/mappings.ts:41]: this change scares me, I would love to leave this all as is and fix it in the one spot where it causes problems

File: components/contract-components/contract-deploy-form/built-in-contract.tsx [hunk 2]
```diff
@@ -265,7 +265,7 @@ const BuiltinContractForm: React.FC<BuiltinContractFormProps> = ({
               });
               onSuccess();
               router.push(
-                `/${SupportedChainIdToNetworkMap[selectedChain]}/${UrlMap[contractType]}/${contractAddress}`,
+                `/${SupportedChainIdToNetworkMap[selectedChain]}${UrlMap[contractType]}/${contractAddress}`,
```
Comment (jnsdls) [inline components/contract-components/contract-deploy-form/built-in-contract.tsx:268]: could we just fix it in this spot? could we just do a `.replaceAll("//", "/")` and be done with it?

File: pages/dashboard/index.tsx [hunk 3]
```diff
@@ -862,7 +862,7 @@ const AsyncContractNameCell: React.FC<AsyncContractNameCellProps> = ({
   );
 
   const contractTypeUrlSegment =
-    cell.contractType === "custom" ? "" : `/${UrlMap[cell.contractType]}`;
+    cell.contractType === "custom" ? "" : `${UrlMap[cell.contractType]}`;
```
Comment (jnsdls) [inline pages/dashboard/index.tsx:865]: if we can solve it the other way then we can get rid of this change

File: constants/mappings.ts (hunk 1 above)
Comment (jnsdls) [reply constants/mappings.ts:41]: scares me because I don't have a good overview of where else this might be used if that makes sense

**Review by jnsdls — CHANGES_REQUESTED** [review]
small fixes

**Comment by nachoiacovino** [comment]
we would need to fix it or do the replace in different places @jnsdls, we can do that definitely but it made more sense to me like this.

**Review by jnsdls — APPROVED** [review]
LGTM, @nachoiacovino make sure it all works, otherwise u fix it

**Pull request merged**